sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2562,18 +2562,17 @@ case class ToBinary(expr: Expression, format: Option[Expression], child: Express
 
   def this(expr: Expression, format: Expression) = this(expr, Option(format),
     format match {
-      case lit if lit.foldable =>
+      case lit if (lit.foldable && Seq(StringType, NullType).contains(lit.dataType)) =>
```

Comment:
BTW, do other systems like snowflake support `to_binary('abc', null)`?

Reply:
Snowflake doesn't specify that in their document, but BigQuery and Teradata support `to_binary('abc', null)`:

[Expressions, functions, and operators | BigQuery | Google Cloud](https://cloud.google.com/bigquery/docs/reference/standard-sql/functions-and-operators#format_string_as_bytes) "If an operand is NULL, the result is NULL, with the exception of the IS operator."

[Teradata Online Documentation | Quick access to technical manuals](https://docs.teradata.com/r/kmuOwjp1zEYg98JsB8fu_A/etRo5aTAY9n5fUPjxSEynw) "If either instring or in_encoding is NULL, the result is NULL."